smriprep/workflows/surfaces.py

```diff
@@ -239,39 +247,82 @@ def init_surface_recon_wf(
         name='sync',
     )
 
+    if not fs_no_resume:
```

Comment:
Large parts of this block are duplicated from above.

1) Create all the nodes and only stick connect statements in the `if`/`else` block
2) Create all the unconditional nodes outside the `if`/`else` block, and the conditional nodes inside